awx/main/management/commands/inventory_import.py

```diff
@@ -959,34 +904,100 @@ def mark_org_limits_failure(self, save=True):
         self.inventory_update.save(update_fields=['org_host_limit_error'])
 
     def handle(self, *args, **options):
-        self.verbosity = int(options.get('verbosity', 1))
-        self.set_logging_level()
-        self.inventory_name = options.get('inventory_name', None)
-        self.inventory_id = options.get('inventory_id', None)
-        venv_path = options.get('venv', None)
+        # Load inventory and related objects from database.
+        inventory_name = options.get('inventory_name', None)
+        inventory_id = options.get('inventory_id', None)
+        if inventory_name and inventory_id:
+            raise CommandError('--inventory-name and --inventory-id are mutually exclusive')
+        elif not inventory_name and not inventory_id:
+            raise CommandError('--inventory-name or --inventory-id is required')
+
+        with advisory_lock('inventory_{}_import'.format(inventory_id)):
```

Comment:
I think that we shouldn't use 2 locks for this, because I don't think it's necessary and it's complicating other considerations.

The bug you fixed came down to the fact that we prematurely initialized the `inventory` object. In this PR, it is passed into the `perform_update` method via `inventory_update.inventory`. We need load inventory after we obtain the lock (so that we get updated `inventory.variables`, in particular), but we can re-load it.

Where we obtain the lock inside of `perform_update`, where we used to do `self.load_inventory_from_database()`, we just instead just do `inventory_update.inventory.refresh_from_db()`, and then that should be plenty to get rid of the outer lock inside of `handle` IMO